File: java/169_MajorityElement.java
//damn slow
/*
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> mp = new HashMap<>();
        int res = 0;
        int curRes = 0;
        for(int num : nums){
          int count = mp.getOrDefault(num, 0) + 1;
          mp.put(num, count);
          if(count > curRes) {
              curRes = count;
              res = num;
          }
        }
        return res;
    }
}
*/

File: c++/169_MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int r = 0, c = 0;
        for (auto x : nums) {
            if (c == 0) {
                r = x, c = 1;
            } else if (r == x) {
                c++;
            } else {
                c--;
            }
        }
        return r;
    }
};